src/libraries/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs

```diff
@@ -135,14 +135,18 @@ internal void ReleaseConnection()
                 {
                     if (!_isClosed && _tcpClient != null)
                     {
-                        //free cbt buffer
-                        if (_channelBindingToken != null)
+                        try
                         {
-                            _channelBindingToken.Close();
+                            // Gracefully close the transmission channel
+                            QuitCommand.Send(this);
+                        }
+                        finally
+                        {
+                            //free cbt buffer
+                            _channelBindingToken?.Close();
```

Comment:
`_channelBindingToken?.Close()` was previously inside of the try/catch to ensure the stream/client get disposed if it fails. Can it no longer fail?

Reply:
> was previously inside of the try/catch 

Old code wasn't inside try/catch I added new one to close in case of quit command exception(if I understood the question).

Reply:
Thanks, I misread the diff there.

Reply:
No worries, let me know if I need to do other on this or if we can merge.